doc/faq.rst

```diff
@@ -0,0 +1,229 @@
+:orphan:
+
+.. _manual:
+
+===
+FAQ
+===
+
+Errors
+======
+
+Auto-sklearn is extremely memory hungry in a sequential setting
+---------------------------------------------------------------
+
+Auto-sklearn can appear very memory hungry (i.e. requiring a lot of memory for small datasets) due
+to the use of ``fork`` for creating new processes when running in sequential manner (if this
+happens in a parallel setting or if you pass your own dask client this is due to a different
+issue).
+
+Let's go into some more detail and discuss how to fix it:
+Auto-sklearn executes each machine learning algorithm in its own process to be able to apply a
+memory limit and a time limit. To start such a process, Python gives three options: ``fork``,
+``forkserver`` and ``spawn``. The default ``fork`` copies the whole process memory into the
+subprocess. If the main process already uses 1.5GB of main memory and we apply a 3GB memory
+limit to Auto-sklearn, executing a machine learning pipeline is limited to use at most 1.5GB.
+We would have loved to use ``forkserver`` or ``spawn`` as the default option instead, which both
+copy only relevant data into the subprocess and thereby alleaviate the issue of eating up a lot
+of your main memory
+(and also do not suffer from potential deadlocks as ``fork`` does, see
+`here <https://pythonspeed.com/articles/python-multiprocessing/>`_),
+but they require the Auto-sklearn code to be guarded by ``if __name__ == "__main__"`` or executed
+in a notebook and we decided that we do not want to require this by default.
+
+There are now two possible solutions:
+
+1. Use Auto-sklearn in parallel: if you use Auto-sklean in parallel, it defaults to ``forkserver``
+   as the parallelization mechanism itself requires Auto-sklearn the code to be guarded. Please
+   find more information on how to do this in the following two examples:
+
+   1. :ref:`sphx_glr_examples_60_search_example_parallel_n_jobs.py`
+   2. :ref:`sphx_glr_examples_60_search_example_parallel_manual_spawning_cli.py`
+
+2. Pass a `dask client <https://distributed.dask.org/en/latest/client.html>`_. If the user passes
+   a dask client, Auto-sklearn can no longer assume that it runs in sequential mode and will use
+   a ``forkserver`` to start new processes.
+
+We therefore suggest using one of the above settings by default.
+
+Auto-sklearn is extremely memory hungry in a parallel setting
+-------------------------------------------------------------
+
+When running Auto-sklearn in a parallel setting it starts new processes for evaluating machine
+learning models using the ``forkserver`` mechanism. If not all code in the main script is guarded
+by ``if __name__ == "__main__"`` it is executed for each subprocess. If such code is for example
```

Comment:
Passive voice makes this sentence hard to understand

Reply:
If not all code in the main script is guarded by ``if __name__ == "__main__"`` it is executed for each subprocess.

-->

Code that is not in the main script and guarded by ``if __name__ == "__main__"`` is executed for each subprocess, e.g. if you are loading your dataset outside of this codeblock it will be executed for each evaluation of a machine learning algorithm and thus blocks your RAM